### PR TITLE
feat(frontend): rename and harmonize the relation between user and space

### PR DESCRIPTION
**Pull request by floreal15**
Refactor of the front and the backend needed after the PR : https://github.com/tracim/tracim/pull/6121

https://github.com/tracim/tracim/issues/6252

The purpose of this pr is to make a refactor needed by some previous pr, it has to rename some variable to more appropriate name : 
what we usually called role is now called config (because the config of an user in a space is both the role and the email notification type, it's not only the role like it was), 
member became role or config depending on what they are ( config is the full list of config of an user in a space while role is just the role and not the full list of config)

The refactor also change the place where the config are, before the config where in the current workspace, witch mean that an user know the config of all the other user in each workspace they have in common, but now each user will only have his config in workspace list and will only have access to necessary data for the other user in his workspace ( the role of each user is needed but the email notification isn't )

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend/src/action-creator.sync.js [hunk 1]
```diff
@@ -118,35 +113,36 @@ export const updateWorkspaceDetail = workspaceDetail => ({ type: `${UPDATE}/${WO
...
 
-const MEMBER = 'Member'
-export const WORKSPACE_MEMBER = `${WORKSPACE}/${MEMBER}`
-export const WORKSPACE_MEMBER_LIST = `${WORKSPACE_MEMBER}/List`
-export const setWorkspaceMemberList = workspaceMemberList => ({ type: `${SET}/${WORKSPACE_MEMBER_LIST}`, workspaceMemberList })
-// INFO - CH - 2023-11-22 - WORKSPACE_MEMBER_ADD is used by api function postWorkspaceMember()
+const ROLE = 'Role'
+export const USER_ROLE = `${USER}/${ROLE}`
+export const USER_ROLE_LIST = `${USER_ROLE}/List`
+export const setWorkspaceMemberList = workspaceMemberList => ({ type: `${SET}/${USER_ROLE_LIST}`, workspaceMemberList })
```
Comment (Skylsmoi) [inline frontend/src/action-creator.sync.js:119]: rename setWorkspaceMemberList to setUserRoleList = userRoleList =>...
and create a const for `${SET}/${USER_ROLE_LIST}` and use it in reducers

File: frontend/src/action-creator.sync.js [hunk 2]
```diff
@@ -118,35 +113,36 @@ export const updateWorkspaceDetail = workspaceDetail => ({ type: `${UPDATE}/${WO
...
+// INFO - CH - 2023-11-22 - USER_ROLE_ADD is used by api function postWorkspaceMember()
 // but nothing is bound to it
-export const WORKSPACE_MEMBER_ADD = `${WORKSPACE_MEMBER}/${ADD}`
-// INFO - CH - 2023-11-22 - WORKSPACE_MEMBER_ADD is used by api function deleteWorkspaceMember()
+export const USER_ROLE_ADD = `${USER_ROLE}/${ADD}`
```
Comment (Skylsmoi) [inline frontend/src/action-creator.sync.js:122]: this duplicate ADD_USER_ROLE
the action should be the prefix so it's this one that should be removed

Comment (Skylsmoi) [reply frontend/src/action-creator.sync.js:122]: and its usage in postWorkspaceMember should be refactored

File: frontend/src/component/Account/UserSpacesConfig.jsx [hunk 3]
```diff
@@ -122,7 +121,7 @@ export const UserSpacesConfig = (props) => {
         ? {
           ...space,
           memberList: space.memberList.map(member => member.id === data.fields.user.user_id
-            ? { ...member, ...serializeMember({ user: data.fields.user, ...data.fields.member }) }
+            ? { ...member, ...serializeUserConfig({ user: data.fields.user, ...data.fields.member }) }
```
Comment (Skylsmoi) [inline frontend/src/component/Account/UserSpacesConfig.jsx:124]: serializeUserConfig should be renamed to serializeUserWorkspaceConfig

File: frontend/src/reducer/workspaceList.js [hunk 4]
```diff
@@ -28,24 +30,25 @@ export const serializeWorkspaceListProps = {
   members: 'memberList'
 }
 
-export const serializeWorkspace = workspace => {
+export const serializeUserConfig = m => {
```
Comment (Skylsmoi) [inline frontend/src/reducer/workspaceList.js:33]: rename to serializeUserWorkspaceConfig

File: frontend/src/reducer/currentWorkspace.js [hunk 5]
```diff
@@ -97,37 +94,37 @@ export default function currentWorkspace (state = defaultWorkspace, action) {
         sidebarEntryList: action.workspaceDetail.sidebar_entries.map(sbe => serialize(sbe, serializeSidebarEntryProps))
       }
 
-    case `${SET}/${WORKSPACE_MEMBER_LIST}`:
+    case `${SET}/${USER_ROLE_LIST}`:
```
Comment (Skylsmoi) [inline frontend/src/reducer/currentWorkspace.js:97]: add comment explaining that USER_ROLE is an object made from an user and a role

File: frontend/src/reducer/workspaceList.js [hunk 6]
```diff
@@ -28,24 +30,25 @@ export const serializeWorkspaceListProps = {
...
+    emailNotificationType: m.email_notification_type || EMAIL_NOTIFICATION_TYPE.NONE,
+    hasAvatar: m.user.has_avatar || true,
+    hasCover: m.user.has_cover || false
   }
 }
 
-export const serializeRole = role => {
-  // TODO - CH - 2023-11-02 - This function should use serializeWorkspace
-  // and be renamed according to this issue: https://github.com/tracim/tracim/issues/6252
+export const serializeUserWorkspaceConfig = config => {
```
Comment (Skylsmoi) [inline frontend/src/reducer/workspaceList.js:45]: rename config to userWorkspaceConfig

File: frontend/src/reducer/workspaceList.js [hunk 7]
```diff
@@ -58,28 +61,26 @@ function addWorkspaceSetting (setting, user, workspace, workspaceList) {
...
 }
 
 export function workspaceList (state = [], action, lang) {
   switch (action.type) {
-    case `${SET}/${ROLE_WORKSPACE_LIST}`:
-      return action.workspaceList.map(serializeRole)
-
-    case `${SET}/${WORKSPACE_LIST}`:
-      return action.workspaceList.map(serializeWorkspace)
+    // FIXME - F.S. - 2023-11-27 - change test of `${SET}/${WORKSPACE_LIST}` to test `${SET}/${USER_WORKSPACE_CONFIG_LIST}`
```
Comment (Skylsmoi) [inline frontend/src/reducer/workspaceList.js:71]: remove comment

**Review by Skylsmoi — CHANGES_REQUESTED** [review]

File: frontend/src/reducer/workspaceList.js (hunk 4 above)
Comment (floreal15) [reply frontend/src/reducer/workspaceList.js:33]: serializeUserWorkspaceConfig is another function used to seralize userWorkspaceConfig object with user, workspace and config value, but in this case it only serialize a user and it's config

File: frontend/test/fixture/workspace/globalManagerWorkspce.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { globalManagerFromApi } from '../user/globalManagerFromApi.js'
+import { firstWorkspaceFromApi } from './firstWorkspace.js'
+
+export const globalManagerWorkspaceFromApi = {
+  email_notification_type: 'summary',
+  role: 'workspace-manager',
```
[the remark this answers is not in the document]
Comment (floreal15) [reply frontend/test/fixture/workspace/globalManagerWorkspce.js:6]: it's a userWorkspaceConfig, i will rename it as globalManagerWorkspaceConfigFromApi

File: frontend/src/action-creator.sync.js (hunk 1 above)
Comment (floreal15) [reply frontend/src/action-creator.sync.js]: done

File: frontend/src/action-creator.sync.js (hunk 2 above)
Comment (floreal15) [reply frontend/src/action-creator.sync.js]: done

File: frontend/src/reducer/currentWorkspace.js (hunk 5 above)
Comment (floreal15) [reply frontend/src/reducer/currentWorkspace.js]: done

File: frontend/src/reducer/workspaceList.js (hunk 6 above)
Comment (floreal15) [reply frontend/src/reducer/workspaceList.js]: done

File: frontend/src/reducer/workspaceList.js (hunk 7 above)
Comment (floreal15) [reply frontend/src/reducer/workspaceList.js]: done

File: frontend/src/reducer/currentWorkspace.js [hunk 9]
```diff
@@ -64,13 +62,12 @@ export const serializeSidebarEntryProps = {
   label: 'label'
 }
 
-export const serializeMember = m => {
+export const serializeRole = m => {
```
[the remark this answers is not in the document]
Comment (floreal15) [reply frontend/src/reducer/currentWorkspace.js]: done

File: frontend/test/src/reducer/workspaceList.spec.js [hunk 10]
```diff
@@ -53,25 +52,27 @@ describe('workspaceList reducer', () => {
...
       it('should return a workspace list with the new list', () => {
         expect(rez).to.deep.equal([
           {
             ...serializedFirstWorkspaceFromApi,
             sidebarEntryList: firstWorkspaceFromApi.sidebar_entries.map(
               sbe => serialize(sbe, serializeSidebarEntryProps)
-            )
+            ),
+            memberList: [serializeUserConfig(globalManagerAsMemberFromApi)
+            ]
```
[the remark this answers is not in the document]
Comment (floreal15) [reply frontend/test/src/reducer/workspaceList.spec.js]: done

File: frontend/src/action-creator.sync.js (hunk 1 above)
Comment (Skylsmoi) [reply frontend/src/action-creator.sync.js]: the const ${SET}/${USER_ROLE_LIST} hasn't been created

File: frontend/src/action-creator.sync.js [hunk 11]
```diff
@@ -118,35 +113,32 @@ export const updateWorkspaceDetail = workspaceDetail => ({ type: `${UPDATE}/${WO
...
-// but nothing is bound to it
-export const WORKSPACE_MEMBER_REMOVE = `${WORKSPACE_MEMBER}/${REMOVE}`
-export const WORKSPACE_MEMBER_UPDATE = `${WORKSPACE_MEMBER}/${UPDATE}`
-export const ADD_WORKSPACE_MEMBER = `${ADD}/${WORKSPACE_MEMBER}`
+const ROLE = 'Role'
+export const USER_ROLE = `${USER}/${ROLE}`
+export const USER_ROLE_LIST = `${USER_ROLE}/List`
+export const SET_USER_ROLE_LIST = `${SET}/${USER_ROLE_LIST}`
+export const setUserRoleList = userRoleList => ({ type: SET_USER_ROLE_LIST, userRoleList })
+export const USER_ROLE_UPDATE = `${USER_ROLE}/${UPDATE}`
```
Comment (Skylsmoi) [inline frontend/src/action-creator.sync.js:121]: the "update" should be before, it's the same as "WORKSPACE_MEMBER_ADD" that you removed
That const doesn't seem to be used anyway, you can remove it

File: frontend/src/component/Account/UserSpacesConfig.jsx (hunk 3 above)
Comment (Skylsmoi) [reply frontend/src/component/Account/UserSpacesConfig.jsx:124]: wont fix, it's the proper one that is used

File: frontend/src/container/Dashboard.jsx [hunk 12]
```diff
@@ -23,19 +23,19 @@ import {
   buildHeadTitle,
   removeAtInUsername,
   addExternalLinksIcons,
-  getSpaceMemberList
+  getSpaceUserRoleList
```
Comment (Skylsmoi) [inline frontend/src/container/Dashboard.jsx:26]: what is "SpaceUserRoleList"? Shouldn't it be "UserRoleList"?

File: frontend/src/reducer/system.js [hunk 13]
```diff
@@ -24,7 +24,7 @@ export function system (state = defaultSystem, action) {
     case `${SET}/${LOGIN}/Redirect`:
       return { ...state, redirectLogin: action.url }
 
-    case `${SET}/${ROLE_WORKSPACE_LIST}`:
+    case `${SET}/${USER_WORKSPACE_CONFIG_LIST}`:
```
Comment (Skylsmoi) [inline frontend/src/reducer/system.js:27]: a const should be created for this case

File: frontend/src/reducer/workspaceList.js [hunk 14]
```diff
@@ -58,28 +61,25 @@ function addWorkspaceSetting (setting, user, workspace, workspaceList) {
...
 }
 
 export function workspaceList (state = [], action, lang) {
   switch (action.type) {
-    case `${SET}/${ROLE_WORKSPACE_LIST}`:
-      return action.workspaceList.map(serializeRole)
-
-    case `${SET}/${WORKSPACE_LIST}`:
-      return action.workspaceList.map(serializeWorkspace)
+    case `${SET}/${USER_WORKSPACE_CONFIG_LIST}`:
```
Comment (Skylsmoi) [inline frontend/src/reducer/workspaceList.js:71]: the const previously mentioned in a comment should be used here too

File: frontend/src/reducer/workspaceList.js [hunk 15]
```diff
@@ -105,6 +105,14 @@ export function workspaceList (state = [], action, lang) {
       return sortListByMultipleCriteria(spaceList, [SORT_BY.LABEL, SORT_BY.ID], SORT_ORDER.ASCENDING, lang)
     }
 
+    case `${UPDATE}/${USER_WORKSPACE_EMAIL_NOTIFICATION_TYPE}`:
```
Comment (Skylsmoi) [inline frontend/src/reducer/workspaceList.js:108]: I would also like a new const for this action

File: frontend/test/src/container/Tracim.spec.js [hunk 16]
```diff
@@ -68,7 +68,7 @@ describe('<Tracim />', () => {
       case `${SET}/${USER_CONFIGURATION}`:
         setUserConfigurationCallBack()
         break
-      case `${SET}/${ROLE_WORKSPACE_LIST}`:
+      case `${SET}/${USER_WORKSPACE_CONFIG_LIST}`:
```
Comment (Skylsmoi) [inline frontend/test/src/container/Tracim.spec.js:71]: usage for the const here too

File: frontend/test/src/reducer/workspaceList.spec.js [hunk 17]
```diff
@@ -53,25 +53,26 @@ describe('workspaceList reducer', () => {
       expect(rez).to.deep.equal(initialState)
     })
 
-    describe(`${SET}/${WORKSPACE_LIST}`, () => {
-      const rez = workspaceList(initialState, setWorkspaceList([firstWorkspaceFromApi]))
+    describe(`${SET}/${USER_WORKSPACE_CONFIG_LIST}`, () => {
```
Comment (Skylsmoi) [inline frontend/test/src/reducer/workspaceList.spec.js:56]: usage of the const here too

**Review by Skylsmoi — CHANGES_REQUESTED** [review]

File: frontend/src/reducer/workspaceList.js (hunk 4 above)
Comment (Skylsmoi) [reply frontend/src/reducer/workspaceList.js]: agreed

File: frontend/test/fixture/workspace/globalManagerWorkspce.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (Skylsmoi) [reply frontend/test/fixture/workspace/globalManagerWorkspce.js]: agreed

File: frontend/src/container/Dashboard.jsx (hunk 12 above)
Comment (Skylsmoi) [reply frontend/src/container/Dashboard.jsx:26]: as seen together, don't rename it

File: frontend/src/reducer/system.js (hunk 13 above)
Comment (floreal15) [reply frontend/src/reducer/system.js]: done

File: frontend/src/reducer/workspaceList.js (hunk 14 above)
Comment (floreal15) [reply frontend/src/reducer/workspaceList.js]: done

File: frontend/src/reducer/workspaceList.js (hunk 15 above)
Comment (floreal15) [reply frontend/src/reducer/workspaceList.js]: done

File: frontend/test/src/container/Tracim.spec.js (hunk 16 above)
Comment (floreal15) [reply frontend/test/src/container/Tracim.spec.js]: done

File: frontend/test/src/reducer/workspaceList.spec.js (hunk 17 above)
Comment (floreal15) [reply frontend/test/src/reducer/workspaceList.spec.js]: done

File: backend/official_plugins/tracim_backend_child_removal/__init__.py [hunk 18]
```diff
@@ -16,14 +16,14 @@ class ChildRemovalPlugin:
     """
 
     @hookimpl
-    def on_user_role_in_workspace_deleted(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_deleted(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/official_plugins/tracim_backend_child_removal/__init__.py:20]: "config" is too ambiguous, the concept is user_workspace_config so it's better to use its full name

File: backend/official_plugins/tracim_backend_parent_access/__init__.py [hunk 19]
```diff
@@ -15,22 +15,22 @@ class ParentAccessPlugin:
...
                 and obj.workspace.workspace_id == workspace_id
             ):
                 return True
         return False
 
     @hookimpl
-    def on_user_role_in_workspace_created(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_created(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/official_plugins/tracim_backend_parent_access/__init__.py:27]: same comment about config

File: backend/tracim_backend/applications/agenda/lib.py [hunk 20]
```diff
@@ -738,19 +738,19 @@ def on_user_deleted(self, user: User, context: TracimContext) -> None:
             self.sync_workspace_symlinks(role, context)
 
     @hookimpl
-    def on_user_role_in_workspace_deleted(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_deleted(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/applications/agenda/lib.py:742]: same comment about config

File: backend/tracim_backend/applications/agenda/lib.py [hunk 21]
```diff
@@ -696,7 +696,7 @@ def sync_user_agenda(self, user: User, context: TracimContext, create_event: boo
                 logger.exception(self, exc)
 
     def sync_workspace_symlinks(
-        self, role: UserRoleInWorkspace, context: TracimContext, role_deletion=False
+        self, role: UserConfigInWorkspace, context: TracimContext, role_deletion=False
```
Comment (Skylsmoi) [inline backend/tracim_backend/applications/agenda/lib.py:699]: did not rename param role, is it normal?

File: backend/tracim_backend/applications/agenda/lib.py [hunk 22]
```diff
@@ -738,19 +738,19 @@ def on_user_deleted(self, user: User, context: TracimContext) -> None:
...
+        self, config: UserConfigInWorkspace, context: TracimContext
     ) -> None:
-        self.sync_workspace_symlinks(role=role, context=context, role_deletion=True)
+        self.sync_workspace_symlinks(role=config, context=context, role_deletion=True)
 
     @hookimpl
-    def on_user_role_in_workspace_modified(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_modified(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/applications/agenda/lib.py:748]: same comment about config

File: backend/tracim_backend/applications/agenda/lib.py [hunk 23]
```diff
@@ -738,19 +738,19 @@ def on_user_deleted(self, user: User, context: TracimContext) -> None:
...
     ) -> None:
-        self.sync_workspace_symlinks(role=role, context=context)
+        self.sync_workspace_symlinks(role=config, context=context)
 
     @hookimpl
-    def on_user_role_in_workspace_created(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_created(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/applications/agenda/lib.py:754]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 24]
```diff
@@ -729,29 +729,29 @@ def _create_content_event(
             context=context,
         )
 
-    # UserRoleInWorkspace events
+    # UserConfigInWorkspace events
     @hookimpl
-    def on_user_role_in_workspace_created(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_created(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:735]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 25]
```diff
@@ -729,29 +729,29 @@ def _create_content_event(
...
     ) -> None:
-        self._create_role_event(OperationType.CREATED, role, context)
+        self._create_role_event(OperationType.CREATED, config, context)
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:737]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 26]
```diff
@@ -729,29 +729,29 @@ def _create_content_event(
...
 
     @hookimpl
-    def on_user_role_in_workspace_modified(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_modified(
+        self, config: UserConfigInWorkspace, context: TracimContext
     ) -> None:
-        self._create_role_event(OperationType.MODIFIED, role, context)
+        self._create_role_event(OperationType.MODIFIED, config, context)
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:743]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 27]
```diff
@@ -729,29 +729,29 @@ def _create_content_event(
...
 
     @hookimpl
-    def on_user_role_in_workspace_deleted(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_deleted(
+        self, config: UserConfigInWorkspace, context: TracimContext
     ) -> None:
-        self._create_role_event(OperationType.DELETED, role, context)
+        self._create_role_event(OperationType.DELETED, config, context)
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:749]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 28]
```diff
@@ -729,29 +729,29 @@ def _create_content_event(
...
 
     def _create_role_event(
         self,
         operation: OperationType,
-        role: UserRoleInWorkspace,
+        config: UserConfigInWorkspace,
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:754]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 29]
```diff
@@ -1259,16 +1259,16 @@ def publish_messages_for_event(self, event_id: int) -> None:
 
 class MessageHooks:
     @hookimpl
-    def on_user_role_in_workspaces_created(
-        self, roles: typing.List[UserRoleInWorkspace], context: TracimContext
+    def on_user_config_in_workspaces_created(
+        self, configs: typing.List[UserConfigInWorkspace], context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:1263]: same comment about config

File: backend/tracim_backend/lib/core/event.py [hunk 30]
```diff
@@ -1293,11 +1293,11 @@ def on_workspace_modified(self, workspace: Workspace, context: TracimContext) ->
                 )
 
     @hookimpl
-    def on_user_role_in_workspace_deleted(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_deleted(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/event.py:1297]: same comment about config

File: backend/tracim_backend/lib/core/userworkspace.py [hunk 31]
```diff
@@ -153,64 +153,64 @@ def create_one(
                     user.user_id, workspace.workspace_id
                 )
             )
-        role = UserRoleInWorkspace()
-        role.user = user
-        role.workspace = workspace
-        role.role = role_level
-        role.email_notification_type = email_notification_type
+        setting = UserConfigInWorkspace()
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/userworkspace.py:156]: it's "config" everywhere else (and should be "user_workspace_config")
is there a reason to name it setting here?

File: backend/tracim_backend/lib/crud_hook/caller.py [hunk 32]
```diff
@@ -33,19 +33,19 @@ def _call_hooks(
...
             elif isinstance(obj, Workspace):
                 self._plugin_manager.hook.on_workspace_created(
                     workspace=obj, context=session.context
                 )
-            elif isinstance(obj, UserRoleInWorkspace):
-                self._plugin_manager.hook.on_user_role_in_workspace_created(
-                    role=obj, context=session.context
+            elif isinstance(obj, UserConfigInWorkspace):
+                self._plugin_manager.hook.on_user_config_in_workspace_created(
+                    config=obj, context=session.context
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/crud_hook/caller.py:46]: same comment about config (I think)

File: backend/tracim_backend/lib/crud_hook/caller.py [hunk 33]
```diff
@@ -64,9 +64,9 @@ def _call_hooks(
                 self._plugin_manager.hook.on_user_call_created(
                     user_call=obj, context=session.context
                 )
-        if roles:
-            self._plugin_manager.hook.on_user_role_in_workspaces_created(
-                roles=roles, context=session.context
+        if configs:
+            self._plugin_manager.hook.on_user_config_in_workspaces_created(
+                configs=configs, context=session.context
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/crud_hook/caller.py:69]: same comment about config (I think)

File: backend/tracim_backend/lib/crud_hook/hookspec.py [hunk 34]
```diff
@@ -51,33 +51,33 @@ def on_workspace_deleted(self, workspace: Workspace, context: TracimContext) ->
         ...
 
     @hookspec
-    def on_user_role_in_workspace_created(
+    def on_user_config_in_workspace_created(
         self,
-        role: UserRoleInWorkspace,
+        config: UserConfigInWorkspace,
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/crud_hook/hookspec.py:56]: same comment about config

File: backend/tracim_backend/lib/mail_notifier/notifier.py [hunk 35]
```diff
@@ -500,7 +500,7 @@ def _build_context_for_content_update(
     def _build_email_body_for_content(
         self,
         mako_template_filepath: str,
-        role: UserRoleInWorkspace,
+        config: UserConfigInWorkspace,
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/mail_notifier/notifier.py:503]: same comment about config

File: backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py [hunk 36]
```diff
@@ -1166,28 +1166,28 @@ def on_content_modified(self, content: Content, context: TracimContext) -> None:
         self.index_user(content.current_revision.owner, context)
 
     @hookimpl
-    def on_user_role_in_workspace_created(
+    def on_user_config_in_workspace_created(
         self,
-        role: UserRoleInWorkspace,
+        config: UserConfigInWorkspace,
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py:1171]: same comment about config

In the entire file

File: backend/tracim_backend/tests/plugins/test_child_removal_plugin.py [hunk 37]
```diff
@@ -7,23 +7,23 @@
...
 from tracim_backend.models.roles import WorkspaceRoles
 from tracim_backend.tests.fixtures import *  # noqa:F401,F403
 
 
 class RemoveFromAllSpacesPlugin:
     @hookimpl
-    def on_user_role_in_workspace_deleted(
-        self, role: UserRoleInWorkspace, context: TracimContext
+    def on_user_config_in_workspace_deleted(
+        self, config: UserConfigInWorkspace, context: TracimContext
```
Comment (Skylsmoi) [inline backend/tracim_backend/tests/plugins/test_child_removal_plugin.py:18]: same comment about config

**Review by Skylsmoi — CHANGES_REQUESTED** [review]

File: backend/tracim_backend/tests/plugins/test_child_removal_plugin.py (hunk 37 above)
Comment (floreal15) [reply backend/tracim_backend/tests/plugins/test_child_removal_plugin.py]: done

File: backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py (hunk 36 above)
Comment (floreal15) [reply backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py]: done

File: backend/tracim_backend/lib/mail_notifier/notifier.py (hunk 35 above)
Comment (floreal15) [reply backend/tracim_backend/lib/mail_notifier/notifier.py]: done

File: backend/tracim_backend/lib/crud_hook/hookspec.py (hunk 34 above)
Comment (floreal15) [reply backend/tracim_backend/lib/crud_hook/hookspec.py]: done

File: backend/tracim_backend/lib/crud_hook/caller.py (hunk 33 above)
Comment (floreal15) [reply backend/tracim_backend/lib/crud_hook/caller.py]: done

File: backend/tracim_backend/applications/agenda/lib.py (hunk 21 above)
Comment (floreal15) [reply backend/tracim_backend/applications/agenda/lib.py]: it have been changed by user_workspace_config

File: backend/tracim_backend/applications/agenda/lib.py (hunk 20 above)
Comment (floreal15) [reply backend/tracim_backend/applications/agenda/lib.py]: done

File: backend/official_plugins/tracim_backend_child_removal/__init__.py (hunk 18 above)
Comment (floreal15) [reply backend/official_plugins/tracim_backend_child_removal/__init__.py]: yes it has been changed

File: backend/tracim_backend/views/core_api/workspace_controller.py [hunk 38]
```diff
@@ -297,14 +297,14 @@ def workspaces_members(
             workspace=request.current_workspace
         )
         return [
-            user_workspace_config_api.get_user_role_workspace_with_context(user_config)
+            user_workspace_config_api.get_user_config_workspace_with_context(user_config)
```
[the remark this answers is not in the document]
Comment (floreal15) [reply backend/tracim_backend/views/core_api/workspace_controller.py]: yes it has been done

File: backend/tracim_backend/applications/agenda/lib.py (hunk 22 above)
Comment (floreal15) [reply backend/tracim_backend/applications/agenda/lib.py]: done

File: backend/tracim_backend/applications/agenda/lib.py (hunk 23 above)
Comment (floreal15) [reply backend/tracim_backend/applications/agenda/lib.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 24 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 25 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 26 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 27 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 28 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/event.py (hunk 29 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/event.py]: done

File: backend/tracim_backend/lib/core/userworkspace.py (hunk 31 above)
Comment (floreal15) [reply backend/tracim_backend/lib/core/userworkspace.py]: yes it has been changed by user_workspace_config

File: backend/tracim_backend/lib/crud_hook/caller.py (hunk 32 above)
Comment (floreal15) [reply backend/tracim_backend/lib/crud_hook/caller.py]: done

File: backend/tracim_backend/tests/plugins/test_child_removal_plugin.py (hunk 37 above)
Comment (floreal15) [reply backend/tracim_backend/tests/plugins/test_child_removal_plugin.py]: done

File: backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py (hunk 36 above)
Comment (floreal15) [reply backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py]: done

File: backend/tracim_backend/lib/crud_hook/hookspec.py (hunk 34 above)
Comment (floreal15) [reply backend/tracim_backend/lib/crud_hook/hookspec.py]: done

File: backend/tracim_backend/lib/mail_notifier/notifier.py (hunk 35 above)
Comment (floreal15) [reply backend/tracim_backend/lib/mail_notifier/notifier.py]: done

File: backend/tracim_backend/views/core_api/workspace_controller.py (hunk 38 above)
[the remark this answers is not in the document]
Comment (floreal15) [reply backend/tracim_backend/views/core_api/workspace_controller.py]: done

File: backend/official_plugins/tracim_backend_child_removal/__init__.py (hunk 18 above)
Comment (floreal15) [reply backend/official_plugins/tracim_backend_child_removal/__init__.py]: done

File: backend/official_plugins/tracim_backend_parent_access/__init__.py (hunk 19 above)
Comment (floreal15) [reply backend/official_plugins/tracim_backend_parent_access/__init__.py]: done

File: backend/tracim_backend/applications/agenda/lib.py (hunk 20 above)
Comment (floreal15) [reply backend/tracim_backend/applications/agenda/lib.py]: done

File: frontend/src/action-creator.sync.js (hunk 11 above)
Comment (Skylsmoi) [reply frontend/src/action-creator.sync.js]: const is deleted

File: backend/tracim_backend/applications/agenda/lib.py [hunk 39]
```diff
@@ -696,15 +696,20 @@ def sync_user_agenda(self, user: User, context: TracimContext, create_event: boo
                 logger.exception(self, exc)
 
     def sync_workspace_symlinks(
-        self, role: UserConfigInWorkspace, context: TracimContext, role_deletion=False
+        self,
+        user_workspace_config: UserConfigInWorkspace,
```
Comment (Skylsmoi) [inline backend/tracim_backend/applications/agenda/lib.py:700]: check if we can rename the class UserConfigInWorkspace to UserWorkspaceConfig

**Review by Skylsmoi — CHANGES_REQUESTED** [review]
Last check / change and I'll approve

File: backend/tracim_backend/applications/agenda/lib.py (hunk 39 above)
Comment (floreal15) [reply backend/tracim_backend/applications/agenda/lib.py]: done

**Review by Skylsmoi — APPROVED** [review]

**Comment by Skylsmoi** [comment]
tested on branch

**Pull request merged**